server/src/packages/rwcds-form/rwcds-form.service.ts

```diff
@@ -143,6 +143,27 @@ export class RwcdsFormService {
       &$expand=dcp_ZoningResolution
     `);
 
+    // dcp_includezoningtextamendment is a field on the rwcds-form that should be "Yes" (717170000) if...
+    // there exists a "ZR" action and that action's dcp_ZoningResolution.dcp_zoningresolution is "AppendixF"
+    const zrAppendixF = projectActions.filter(action =>
+      action['_dcp_action_value@OData.Community.Display.V1.FormattedValue'] === 'ZR'
+      && action.dcp_ZoningResolution.dcp_zoningresolution === 'AppendixF'
```

Comment:
not ideal to have static values here, but it does seem this is super one-off and unlikely to surface elsewhere (I hope). Otherwise I would also vouch for transforming `ZONING_RESOLUTION_TYPES` into an optionset and using it here for `ZR`. I don't know if it would be worth it to create one for AppendixF.

Reply:
Can you explain how using an optionset be any different than using the static value? We'd still have to include 'ZR' somewhere to denote that that's the action we want, whether we're using an optionset or not.

Reply:
In my mind it seems more proper to use codes like 71710000 to do business logic comparisons, since that's the language in which computers and CRM speaks. In this specific case I guess it should be fine since action codes are unique and not very messy ('ZR' vs 'ZA' vs 'UK', etc), but sometimes when using human readable labels it gets messy. E.g. maybe it would be 'ZR - Zoning Resolution` or 'ZA - Zoning Action' where there's more potential for whitespace, capitalization or pluralization errors. Also coding static values means those errors would be silent. If someone typed 'ZRS' instead of 'ZR', it would fail silently.

Reply:
I guess my confusion comes from that 'ZR' is how the data is actually stored in CRM. `action['_dcp_action_value@OData.Community.Display.V1.FormattedValue']` is stored as text strings, e.g. `ZR`, `ZC`, `ZA` -- so why would I use a code?

Reply:
Using `optionset` centralizes the use of static lookup values like this. This sort of thing should at least be kept in a named lookup so we avoid littering our code with lookup strings (that might need to change later).

I could see a nice lookup table, even at the top of the file (if not defined and then exported in the optionset helper) looking like this:
```javascript
const ACTION_LOOKUPS = {
  ZONING_RESOLUTION: 'ZR',
  // others ...
};
```

Two reasons for this:

1. clearer, self-documenting code
2. if values changes for some reason, we go to one place
3. if values change for some reason, we don't have to edit those values in multiple places

Reply:
I mean you do the same thing with `INCLUDE_ZONING_TEXT_AMENDMENT_OPTIONSET` which is awesome. Why not keep doing it for these other things? The way you handle `INCLUDE_ZONING_TEXT_AMENDMENT_OPTIONSET` is perfect.

Reply:
Okay I'll turn ZONING_RESOLUTION_TYPES lookup into optionset and then just use it as `ZONING_RESOLUTION_TYPE['ZONING_TEXT_AMENDMENT'].label` (which would equal "ZR")

Reply:
> I guess my confusion comes from the fact that 'ZR' is how the data is actually stored in CRM. `action['_dcp_action_value@OData.Community.Display.V1.FormattedValue']` is stored as text strings, e.g. `ZR`, `ZC`, `ZA` -- so why would I use a code?

@trbmcginnis  Oh, when I saw the optionset I assumed in CRM it is stored as codes:
```
![image](https://user-images.githubusercontent.com/3311663/88964860-d3c6c780-d25e-11ea-9bd7-dfbe4b5d1bc1.png)
```
I think maybe when you specifically access the `...Display.V1.FormattedValue` version of the property, CRM nicely provides the humna-readable label version for you, but it's not the true underlying value

Reply:
Ah, yeah — it's tricky. Basically this is a way to get around having to do another expand. CRM Web API provides a general "formatted value" for a related entity. In this case, the related entity is `dcp_action`, and named formatted value is derived from the action type value.

There are other places we do that in the code.

Reply:
So that lookup is for the dcp_affectedzoningresolution entity, the 'ZR' that I'm using is for the dcp_projectaction entity. Basically the same attribute but on two different entities and with two different entity names.

That's a good point about the formatted value version of the attribute, I didn't notice that. Going to try to look into that and see if it uses codes similar to dcp_affectedzoningresolution